cv_next/app/feed/components/CVItemRSC.tsx

```diff
@@ -1,61 +1,117 @@
+import {
+  getImageURL,
+  revalidatePreview,
+  getUserName,
+} from "@/app/actions/cvs/preview";
+import Definitions from "@/lib/definitions";
 
-const getBlur = cache(async (imageURL: string) => {
-  return await getBlurredCv(imageURL);
-});
 interface CVCardProps {
   cv: CvModel;
 }
 
+const getBlur = cache(async (imageURL: string) => {
+  return await getBlurredCv(imageURL);
+});
+
+const getURL = cache(async (cvId: string) => {
+  return await getImageURL(cvId);
+});
+
+const getCachedUserName = cache(async (userId: string) => {
+  return await getUserName(userId);
+});
+
 export default function CVItemRSC({ cv }: CVCardProps) {
+  const [realURL, setRealURL] = useState("");
+  const [authorName, setAuthorName] = useState("");
+
   const [base64Data, setBase64Data] = useState(
-    "data:iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7p5fDwAFlAI2LB7hbAAAAABJRU5ErkJggg=="
+    "data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7p5fDwAFlAI2LB7hbAAAAABJRU5ErkJggg=="
   );
 
   const categoryLink = `/feed?category=${Categories.category[
     cv.category_id
   ].toLowerCase()}`;
-  const imageUrl = generateImageUrl(cv.document_link);
-  const formattedDate = new Date(cv.created_at).toLocaleDateString("en-US");
 
   useEffect(() => {
-    (async () => {
-      const base64 = await getBlur(imageUrl);
+    const revalidateImage = async () => {
+      await revalidatePreview(cv.document_link);
+      const imageUrl =
+        (await getURL(getIdFromLink(cv.document_link) || "")) ?? "";
+      setRealURL(imageUrl);
+    };
+    const getAuthorName = async () => {
+      const userUploading = (await getCachedUserName(cv.user_id || "")) || "";
+      setAuthorName(userUploading);
+    };
+    const getBlurCv = async () => {
+      const base64 = await getBlur(getGoogleImageUrl(cv.document_link));
       setBase64Data(base64);
-    })();
-  }, [imageUrl]);
+    };
+    getBlurCv();
+    getAuthorName();
+    revalidateImage(); // Immediately invoke revalidateImage
+
+    const interval = setInterval(
+      revalidateImage,
+      Definitions.FETCH_WAIT_TIME * 1000
+    ); // Revalidate every 2 minutes
+
+    return () => clearInterval(interval);
+  }, [cv]);
+
+  const formattedDate = new Date(cv.created_at).toLocaleDateString("en-US");
 
   return (
     <>
-      <Image
-        width={500}
-        height={500 * 1.4142}
-        className="w-full rounded-lg p-2"
-        src={imageUrl}
-        placeholder="blur"
-        blurDataURL={base64Data}
-        alt="CV Preview"
-        priority
-      />
+      {realURL ? (
+        <Image
+          width={500}
+          height={500 * 1.4142}
+          className="w-full rounded-lg p-2"
+          src={realURL}
+          placeholder="blur"
+          blurDataURL={base64Data}
+          alt="CV Preview"
+          priority
+        />
+      ) : (
+        <div style={{ width: "380px", height: `400px`, overflow: "hidden" }}>
```

Comment:
Why change width and height via `style` when there are specific attributes for that in the `Image` component?

Reply:
This is mainly for the initial load(a default valid value must be given to the Image for it to load properly this makes sure that the default data spreads to the entire space instead of only taking what it needs)